tests/query_workflow_test.go

```diff
@@ -257,3 +258,41 @@ func (s *clientIntegrationSuite) TestQueryWorkflow_QueryBeforeStart() {
 	// wait query
 	wg.Wait()
 }
+
+func (s *clientIntegrationSuite) TestQueryWorkflow_QueryFailedWorkflowTask() {
+
+	workflowFn := func(ctx workflow.Context) (string, error) {
+		err := workflow.SetQueryHandler(ctx, "test", func() (string, error) {
+			return "", nil
+		})
+
+		if err != nil {
+			s.Logger.Fatal("SetQueryHandler failed: " + err.Error())
+		}
+		// force workflow task to fail
+		panic("Workflow failed")
+	}
+
+	s.worker.RegisterWorkflow(workflowFn)
+
+	id := "test-query-failed-workflow-task"
+	workflowOptions := sdkclient.StartWorkflowOptions{
+		ID:                 id,
+		TaskQueue:          s.taskQueue,
+		WorkflowRunTimeout: 20 * time.Second,
+	}
+	ctx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
+	defer cancel()
+	workflowRun, err := s.sdkClient.ExecuteWorkflow(ctx, workflowOptions, workflowFn)
+	if err != nil {
+		s.Logger.Fatal("Start workflow failed with err", tag.Error(err))
+	}
+
+	s.NotNil(workflowRun)
+	s.True(workflowRun.GetRunID() != "")
+
+	// wait for workflow to fail
+	time.Sleep(time.Second * 5)
```

Comment:
nit: do we have to wait for 5 sec? seems a bit long. I think workflow task attempt will be larger than 1 as soon as the first workflow task is failed.